x-pack/plugins/maps/public/classes/styles/vector/vector_style.tsx

```diff
@@ -721,13 +721,18 @@ export class VectorStyle implements IVectorStyle {
   };
 
   async hasLegendDetails() {
+    if(this._source.hasLegendDetails){
+      return await this._source.hasLegendDetails()
+    }
     return this._getLegendDetailStyleProperties().length > 0;
   }
 
   renderLegendDetails() {
     const symbolId = this._getSymbolId();
     const svg = symbolId ? this.getIconSvg(symbolId) : undefined;
-
+    if(this._source.renderLegendDetails){
```

Comment:
We can update https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/classes/styles/vector/vector_style.tsx#L113 to be `ReactElement | null`.

Then this can just be 
```
return this._source.renderLegendDetails
  ? this._source.renderLegendDetails(this)
  : <VectorStyleLegend/>
```